src/public/publicAPIs.ts

```diff
@@ -312,4 +312,25 @@ export function navigateToTab(tabInstance: TabInstance, onComplete?: (status: bo
 
   const errorMessage = "Invalid internalTabInstanceId and/or channelId were/was provided";
   GlobalVars.callbacks[messageId] = onComplete ? onComplete : getGenericOnCompleteHandler(errorMessage);
-}
\ No newline at end of file
+}
+
+/**
+ * @private
+ * Hide from docs until release.
+ * ------
+ * 
+ * Sends query to bot in order to retrieve data.
+ * @param botRequest query to send to bot.
+ * @param onBotResponse callback to invoke when data is retrieved from bot
+ */
+export function sendBotRequest(botRequest: BotRequest, onBotResponse?: (status: boolean, data: string | BotResponse) => void): void { // void for now
```

Comment:
Lets add two handler onBotResponse(date) and onError(string) => {}

Reply:
sure